client/app/bundles/course/assessment/submission/components/ScribingView/ToolDropdown.jsx

```diff
@@ -40,44 +42,89 @@ const style = {
     fontSize: '12px',
     padding: '10px 0px 10px 0px',
   },
+  disabled: {
+    cursor: 'not-allowed',
+    pointerEvents: 'none',
+    color: '#c0c0c0',
+  },
 };
 
 export default class ToolDropdown extends Component {
   renderIcon() {
-    const { iconClassname, currentTool, toolType, iconComponent } = this.props;
+    const { disabled, iconClassname, currentTool, toolType, iconComponent } = this.props;
+    let iconStyle;
+    if (disabled) {
+      iconStyle = style.disabled;
+    } else if (currentTool === toolType) {
+      iconStyle = { color: blue500 };
+    } else {
+      iconStyle = { color: 'rgba(0, 0, 0, 0.4)' };
+    }
 
     return iconComponent ?
       iconComponent() :
       <FontIcon
         className={iconClassname}
-        style={
-          currentTool === toolType ?
-            { color: blue500 } :
-            { color: 'rgba(0, 0, 0, 0.4)' }
-        }
+        style={iconStyle}
       />;
   }
 
   renderColorBar() {
-    const { colorBar, colorBarComponent } = this.props;
+    const { activeObject, disabled, colorBarBorder, colorBarBackground } = this.props;
+
+    let backgroundColor = colorBarBackground;
+    let borderColor = colorBarBorder;
+
+    if (activeObject) {
+      switch (activeObject.type) {
+        case 'path':
+        case 'line':
+          backgroundColor = activeObject.stroke;
+          break;
+        case 'i-text':
+          backgroundColor = activeObject.fill;
+          break;
+        case 'rect':
+        case 'ellipse':
+          backgroundColor = activeObject.fill;
+          borderColor = activeObject.stroke;
+          break;
+        default:
+      }
+    }
+
 
-    return colorBarComponent ?
-      colorBarComponent() :
-      <div style={{ ...style.colorBar, background: colorBar }} />;
+    return disabled ?
+      <div
+        style={{
+          width: '23px',
+          height: '8px',
+          background: '#c0c0c0',
+        }}
+      />
+      :
+      <div
+        style={{
+          width: '23px',
+          height: '8px',
```

Comment:
Cleaner to write the following? :

```
const colorBarStyle = ...
return <div style={colorBarStyle} />
```

Can make use of `styles.colorBar` to compute `colorBarStyle` also.